include/nbl/video/SPhysicalDeviceLimits.h

```diff
@@ -0,0 +1,497 @@
+    //VkDeviceSize          bufferImageGranularity;
+    //VkDeviceSize          sparseAddressSpaceSize;
+    //uint32_t              maxBoundDescriptorSets;
+    //uint32_t              maxPerStageDescriptorSamplers;
+    //uint32_t              maxPerStageDescriptorUniformBuffers;
+    uint32_t maxPerStageDescriptorSSBOs;
+    //uint32_t              maxPerStageDescriptorSampledImages;
+    //uint32_t              maxPerStageDescriptorStorageImages;
+    //uint32_t              maxPerStageDescriptorInputAttachments;
+    //uint32_t              maxPerStageResources;
+    //uint32_t              maxDescriptorSetSamplers;
+    uint32_t maxDescriptorSetUBOs;
+    uint32_t maxDescriptorSetDynamicOffsetUBOs;
+    uint32_t maxDescriptorSetSSBOs;
+    uint32_t maxDescriptorSetDynamicOffsetSSBOs;
+    uint32_t maxDescriptorSetImages;
+    uint32_t maxDescriptorSetStorageImages;
+    //uint32_t              maxDescriptorSetInputAttachments;
+    //uint32_t              maxVertexInputAttributes;
+    //uint32_t              maxVertexInputBindings;
+    //uint32_t              maxVertexInputAttributeOffset;
+    //uint32_t              maxVertexInputBindingStride;
+    //uint32_t              maxVertexOutputComponents;
+    //uint32_t              maxTessellationGenerationLevel;
+    //uint32_t              maxTessellationPatchSize;
+    //uint32_t              maxTessellationControlPerVertexInputComponents;
+    //uint32_t              maxTessellationControlPerVertexOutputComponents;
+    //uint32_t              maxTessellationControlPerPatchOutputComponents;
+    //uint32_t              maxTessellationControlTotalOutputComponents;
+    //uint32_t              maxTessellationEvaluationInputComponents;
+    //uint32_t              maxTessellationEvaluationOutputComponents;
+    //uint32_t              maxGeometryShaderInvocations;
+    //uint32_t              maxGeometryInputComponents;
+    //uint32_t              maxGeometryOutputComponents;
+    //uint32_t              maxGeometryOutputVertices;
+    //uint32_t              maxGeometryTotalOutputComponents;
+    //uint32_t              maxFragmentInputComponents;
+    //uint32_t              maxFragmentOutputAttachments;
+    //uint32_t              maxFragmentDualSrcAttachments;
+    //uint32_t              maxFragmentCombinedOutputResources;
+    uint32_t maxComputeSharedMemorySize;
+    //uint32_t              maxComputeWorkGroupCount[3];
+    //uint32_t              maxComputeWorkGroupInvocations;
+    uint32_t maxWorkgroupSize[3];
+    //uint32_t              subPixelPrecisionBits;
+    //uint32_t              subTexelPrecisionBits;
+    //uint32_t              mipmapPrecisionBits;
+    //uint32_t              maxDrawIndexedIndexValue;
+    uint32_t maxDrawIndirectCount;
+    //float                 maxSamplerLodBias;
+    float    maxSamplerAnisotropyLog2;
+    uint32_t maxViewports;
+    uint32_t maxViewportDims[2];
+    //float                 viewportBoundsRange[2];
```

Comment:
there's no query for this in GL, but
> this range must be at least [-2 × size, 2 × size - 1], where `size = max(maxViewportDimensions[0], maxViewportDimensions[1])`.

Also change the type to `uint32_t`

Reply:
why not `int32_t`? as you can see value can be negative (-2 * size)

Reply:
No, but the default value for GL(min supported in VK) has the same value for both dims, the values can be negative because:
> The intent of the viewportBoundsRange limit is to allow a maximum sized viewport to be arbitrarily shifted relative to the output target as long as at least some portion intersects.

Reply:
Yep, [0] is min and [1] is max,
Also it can be floating point

Reply:
> Yep, [0] is min and [1] is max,

both X and Y have the same ranges XD?

Reply:
Seems like it 
> viewportBoundsRange[2] is the [minimum, maximum] range that the corners of a viewport must be contained in.

Reply:
So @devshgraphicsprogramming Should I keep the type `float` same as vulkan? I don't know why you first suggested to convert it to integert

Reply:
> I don't know why you first suggested to convert it to integert

cause I didn't know all the above.

Keep it all as is, just derive the proper value for GL